### DISCUSSION TITLE
How to run Rod and view the browser with docker compose?

### DISCUSSION THREAD

**Discussion by Nikola-Milovic** [go-rod/rod#717] [Q&A]
Hello, I have a question, for development I have docker compose that runs my different services, and for development I'd like be able to view the browser.I tried several ideas, the least painless one was to run a managed server on the host and connect to that, or just have a chromium instance listening in debug mode, but I cannot get either to work. Are there any better approaches?

```
func getBrowser() *rod.Browser {
	l := launcher.MustNewManaged("host.docker.internal:7318")

	// Launch with headful mode
	l.Headless(false).XVFB("--server-num=5", "--server-args=-screen 0 1600x900x16")

	browser := rod.New().Client(l.MustClient()).MustConnect()

	return browser
}
```

And my server thats running

```

var addr = flag.String("address", ":7318", "the address to listen to")

func main() {
	flag.Parse()
	
	m := launcher.NewManager()
	l, err := net.Listen("tcp", *addr)
	if err != nil {
		utils.E(err)
	}

	fmt.Println("rod-manager listening on:", l.Addr().String())
	
	srv := &http.Server{Handler: m}
	utils.E(srv.Serve(l))
}
```

But I get the following error 

```
panic: Get "host.docker.internal:7318": unsupported protocol scheme "host.docker.internal"
```

The problem is I have to go roundabout way of connecting to host as I have the bridge network mode and I cannot change it to host as I need access to other internal services.

**Comment by ysmood** [comment]
[Forward host port to docker container](https://stackoverflow.com/a/18646069)
